app/services/reporting/account_reuse_report.rb

```diff
@@ -39,77 +63,266 @@ def account_reuse_emailable_report
     end
 
     def stats_month
-      report_date.strftime('%b-%Y')
+      report_date.prev_month(1).strftime('%b-%Y')
     end
 
     private
 
-    def total_reuse_report
-      return @total_reuse_report if defined?(@total_reuse_report)
-      reuse_stats = agency_reuse_results
+    ReuseDetailRow = Struct.new(
+      :num_entities, :num_all_users, :all_percent, :num_idv_users,
+      :idv_percent
+    ) do
+      def initialize(num_all_users: 0, all_percent: 0, num_idv_users: 0, idv_percent: 0)
+        super(num_all_users:, all_percent:, num_idv_users:, idv_percent:)
+      end
 
-      reuse_total_users = 0
-      reuse_total_percentage = 0
+      def update_details(num_entities: nil, num_idv_users: nil, num_all_users: nil)
+        self.num_entities = num_entities if !num_entities.nil?
 
-      total_proofed = num_active_profiles
+        self.num_idv_users = num_idv_users if !num_idv_users.nil?
+
+        self.num_all_users = num_all_users if !num_all_users.nil?
+
+        self
+      end
+
+      self
+    end
+
+    ReuseDetailSection = Struct.new(:detail_rows) do
+      def initialize(detail_rows: [ReuseDetailRow.new])
+        super(detail_rows:)
+      end
+
+      def organize_results(all_results, idv_results, entity_type)
+        idv_results.each do |result|
+          entity_num = result["num_#{entity_type}"]
+          self.detail_rows[entity_num] =
+            ReuseDetailRow.new.update_details(
+              num_entities: entity_num,
+              num_idv_users: result['num_idv_users'],
+            )
+        end
+
+        all_results.each do |result|
+          entity_num = result["num_#{entity_type}"]
+          if self.detail_rows[entity_num].is_a?(Struct)
+            self.detail_rows[entity_num].update_details(num_all_users: result['num_all_users'])
+          else
+            self.detail_rows[entity_num] =
+              ReuseDetailRow.new.update_details(
+                num_entities: entity_num,
+                num_all_users: result['num_all_users'],
+              )
+          end
+        end
 
-      if !reuse_stats.empty?
-        reuse_stats.each do |result_entry|
-          reuse_total_users += result_entry['num_users']
+        if self.detail_rows.length > 1
+          # If there are results, then remove the zero placeholder
+          self.detail_rows[0] = nil
+          self.detail_rows = self.detail_rows.compact
```

Comment:
we could probably just do
```suggestion
          # If there are results, then remove the zero placeholder
          detail_rows.delete_at(0)
```

Reply:
Actually, the compact will also remove any `nil` entries in the middle of the array, as well - for example, if there are users who have 2 SPs and 4 SPs, but no users with 3 SPs, then the 3-index of the array would be `nil`.  Does that make sense?

Reply:
seems like it would be weird to jump from 2 to 4? but if that is our goal then yes `compact` makes sense

Reply:
Yeah, I'm using the number of (sps or agencies) as an index to collect the information and calculations - it's happening up in `organize_results`...  I can add a variable to call that out better.

Reply:
Done